mypyc/codegen/emitfunc.py

```diff
@@ -438,7 +438,15 @@ def visit_binary_int_op(self, op: BinaryIntOp) -> None:
         dest = self.reg(op)
         lhs = self.reg(op.lhs)
         rhs = self.reg(op.rhs)
-        self.emit_line('%s = %s %s %s;' % (dest, lhs, op.op_str[op.op], rhs))
+        lhs_cast = ""
+        rhs_cast = ""
+        if (is_tagged(op.lhs.type) and is_tagged(op.rhs.type)
+                and op.op in (BinaryIntOp.SLT, BinaryIntOp.SGT,
+                              BinaryIntOp.SLE, BinaryIntOp.SGE)):
+            lhs_cast = "(Py_ssize_t)"
+            rhs_cast = "(Py_ssize_t)"
+        self.emit_line('%s = %s%s %s %s%s;' % (dest, lhs_cast, lhs,
```

Comment:
We'll also need similar checks for unsigned comparisons -- if the types aren't tagged, unsigned comparisons require a cast to the corresponding unsigned type.

Since we aren't using unsigned comparisons yet, it would be good to add unit tests for these.

Reply:
I agree that we should add these checks when the types are int32/int64 and op is unsigned, though we don't have that kind of usages right now so maybe leave it out from this PR would be OK.

I am not sure what kind of unit tests we are adding here. My guess is to ensure this kind of cast is enforced in the generated C code? Probably in `test_emitfunc.py`?

Reply:
`test_emitfunc.py` would be a reasonable location for unit tests.